sklearn/metrics/_plot/precision_recall.py

```diff
@@ -0,0 +1,183 @@
+from .. import average_precision_score
+from .. import precision_recall_curve
+
+from ...utils import check_matplotlib_support
+from ...utils.validation import check_is_fitted
+from ...base import is_classifier
+
+
+class PrecisionRecallDisplay:
+    """Precision Recall visualization.
+
+    It is recommend to use :func:`~sklearn.metrics.plot_precision_recall_curve`
+    to create a visualizer. All parameters are stored as attributes.
+
+    Parameters
+    -----------
+    precision : ndarray of shape (n_thresholds + 1,)
+        Precision values.
+
+    recall : ndarray of shape (n_thresholds + 1,)
+        Recall values.
+
+    average_precision : float
+        Average precision.
+
+    estimator_name : str
+        Name of estimator.
+
+    Attributes
+    ----------
+    line_ : matplotlib Artist
+        Precision recall curve.
+
+    ax_ : matplotlib Axes
+        Axes with precision recall curve.
+
+    figure_ : matplotlib Figure
+        Figure containing the curve.
+    """
+
+    def __init__(self, precision, recall, average_precision, estimator_name):
+        self.precision = precision
+        self.recall = recall
+        self.average_precision = average_precision
+        self.estimator_name = estimator_name
+
+    def plot(self, ax=None, label_name=None, **kwargs):
+        """Plot visualization.
+
+        Extra keyword arguments will be passed to matplotlib's `plot`.
+
+        Parameters
+        ----------
+        ax : Matplotlib Axes, default=None
+            Axes object to plot on. If `None`, a new figure and axes is
+            created.
+
+        label_name : str, default=None
+            Name of precision recall curve for labeling. If `None`, use the
+            name of the estimator.
+
+        **kwargs : dict
+            Keyword arguments to be passed to matplotlib's `plot`.
+
+        Returns
+        -------
+        display : :class:`~sklearn.metrics.PrecisionRecallDisplay`
+            Object that stores computed values.
+        """
+        check_matplotlib_support("PrecisionRecallDisplay.plot")
+        import matplotlib.pyplot as plt
+
+        if ax is None:
+            fig, ax = plt.subplots()
+
+        label_name = self.estimator_name if label_name is None else label_name
+
+        line_kwargs = {
+            "label": "{} (AP = {:0.2f})".format(label_name,
+                                                self.average_precision),
+            "drawstyle": "steps-post"
+        }
+        line_kwargs.update(**kwargs)
+
+        self.line_, = ax.plot(self.recall, self.precision, **line_kwargs)
+        ax.set(xlabel="Recall", ylabel="Precision", ylim=[0.0, 1.05],
```

Comment:
rely on default xlim/ylim?

Reply:
For the, `x`, going without the I think for the y it is kind of important because of the scaling:

With `ylim` explicit set:
![set](https://user-images.githubusercontent.com/5402633/68429029-52cd1a80-017b-11ea-9aee-45fef4695af8.png)

Not set:
![not_set](https://user-images.githubusercontent.com/5402633/68429037-582a6500-017b-11ea-8c6b-1dcd975fe083.png)